### PR TITLE
Feature/loadbalancer

### PR DESCRIPTION
**Pull request by coverbeck**
LoadBalancer for Dockstore

SEAB-645.

Creates an application load balancer in AWS with the following characteristics:

1. Uses an AWS ACM cert
2. Listens on port 443 and forwards to port 80 on EC2
3. Listens on port 8443 and forwards to port 8443 on EC2
4. Listens on port 80 and redirects to port 443 on the LB itself.

After creating the LB, you still need to manually add the instance(s)
to the target group, and update Route53 to point to the LB.

Requires a [compose_setup](https://github.com/dockstore/compose_setup/pull/119) change as well

### CODE REVIEW

File: loadbalancer.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+  # Listen on port 80 and redirect to port 443 on the the same LB
+  LbHtpToHttpsListener:
+    Type: 'AWS::ElasticLoadBalancingV2::Listener'
+    Properties:
+      DefaultActions:
+        - Type: redirect
+          RedirectConfig:
+            Protocol: "HTTPS"
+            Port: "843"
```
Comment (garyluu) [inline loadbalancer.yml]: Is this port and the comment supposed to match?

File: loadbalancer.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      Name: !Sub 'webservice-port-80-${AWS::StackName}'
+      Port: 80
+      Protocol: HTTP
+      VpcId: !Ref VpcId
+  # This target group will receive traffic on port 8443
+  TargetGroupPort8443:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties: 
+      HealthCheckEnabled: true
+      HealthCheckIntervalSeconds: 60
```
Comment (denis-yuen) [inline loadbalancer.yml]: This brings up a thought. Doesn't have to do with this line, but did you test whether load balancers affected by the same issue that we had with CloudFront (long running API calls getting terminated)

i.e. do they hold open a connection between the client and server?
or do they simply forward you to the server and then step away from the process?

File: loadbalancer.yml (hunk 1 above)
Comment (coverbeck) [reply loadbalancer.yml]: D'oh! Fixed

File: loadbalancer.yml (hunk 2 above)
Comment (coverbeck) [reply loadbalancer.yml]: Good catch; the default is idle timeout is 60 seconds but it is configurable up to 4,000 seconds. Adding it and testing it.

File: loadbalancer.yml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  Subnet2:
+    Description: The second existing subnet, which must be in VpcId
+    Type: String
+  CertificateArn:
+    Description: The AWS Certificate Manager ARN
+    Type: String
+  LoadBalancerIdleTimeout:
+    Description: The idle timeout value, in seconds
+    Type: Number
+    Default: 300
```
Comment (denis-yuen) [inline loadbalancer.yml]: What is the reasoning for 300?

Comment (coverbeck) [reply loadbalancer.yml]: I had it my mind that 300 is our current timeout, but it's [actually 600](https://github.com/dockstore/compose_setup/blob/0d9010bf46e2447472bf87dd2aed9dd6483a159d/templates/default.nginx_https.shared.conf.template#L5). Will update.

**Pull request merged**